src/patroni.py

```diff
@@ -70,6 +73,16 @@ def _patroni_url(self) -> str:
         """Patroni REST API URL."""
         return f"{'https' if self._tls_enabled else 'http'}://{self._endpoint}:8008"
 
+    @property
+    def rock_postgresql_version(self) -> Optional[str]:
+        """Version of Postgresql installed in the Rock image."""
+        container = self._charm.unit.get_container("postgresql")
+        if not container.can_connect():
+            logger.debug("Cannot get Postgresql version from Rock. Container inaccessible")
+            return
+        snap_meta = container.pull("/meta.charmed-postgresql/snap.yaml")
+        return yaml.safe_load(snap_meta)["version"]
+
```

Comment:
Version is now needed to set the BIN_DIR in Patroni config. Getting the version specified in the snap manifest file.